manager/apis/app/v1alpha1/fybrikapplication_types.go

```diff
@@ -110,12 +110,29 @@ type ResourceReference struct {
 	AppVersion int64 `json:"appVersion"`
 }
 
+// SecretRef contains the details of a secret
+type SecretRef struct {
+	// Secret Namespace
+	// +required
+	Namespace string `json:"namespace"`
+	// Secret name
+	// +required
+	Name string `json:"name"`
+}
+
 // DatasetDetails holds details of the provisioned storage
 type DatasetDetails struct {
 	// Reference to a Dataset resource containing the request to provision storage
+	// +optional
```

Comment:
Why optional?

Reply:
I thought that `omitempty` means it can be optional.